### PR TITLE
Do not allow `ClassVar` and `Final` in `TypedDict` and `NamedTuple`

### LINKED ISSUE #18220: Disallow `ClassVar` and `Final` qualifiers in `TypedDict` and `NamedTuple` items
Both `ClassVar` and `Final` produce runtime errors when used in the body of a `TypedDict` or `NamedTuple`.

Mypy already emits an error for `Final`, but the error message could be clarified.

```python
class TD(TypedDict):
    a: Final[int]     # E: Final can be only used as an outermost qualifier in a variable annotation
    b: ClassVar[int]  # Runtime error, no mypy error

class NT(NamedTuple):
    a: Final[int]     # E: Final name must be initialized with a value \
                      # E: Final can be only used as an outermost qualifier in a variable annotation
    b: ClassVar[int]  # Runtime error, no mypy error
```

Follows from 
* https://github.com/python/typing-council/issues/39

* https://github.com/microsoft/pyright/issues/9526

### PR DESCRIPTION
**Pull request by sobolevn**
Closes #18220

### CODE REVIEW

File: test-data/unit/check-namedtuple.test [hunk 1]
```diff
@@ -1441,3 +1441,16 @@ def bar() -> None:
...
+
+
+[case testNamedTupleFinalAndClassVar]
+from typing import NamedTuple, Final, ClassVar
+
+class My(NamedTuple):
+    a: Final      # E: Final[...] can't be used inside a NamedTuple
+    b: Final[int] # E: Final[...] can't be used inside a NamedTuple
+    c: ClassVar       # E: ClassVar[...] can't be used inside a NamedTuple
+    d: ClassVar[int]  # E: ClassVar[...] can't be used inside a NamedTuple
```
Comment (JukkaL) [inline test-data/unit/check-namedtuple.test:1453]: What about functional forms, e.g. `My = NamedTuple(...)`? Should we test them as well?

File: test-data/unit/check-typeddict.test [hunk 2]
```diff
@@ -4053,3 +4053,16 @@ d: D = {"a": 1, "b": "x"}
...
+
+
+[case testTypedDictFinalAndClassVar]
+from typing import TypedDict, Final, ClassVar
+
+class My(TypedDict):
+    a: Final      # E: Final[...] can't be used inside a TypedDict
+    b: Final[int] # E: Final[...] can't be used inside a TypedDict
+    c: ClassVar       # E: ClassVar[...] can't be used inside a TypedDict
+    d: ClassVar[int]  # E: ClassVar[...] can't be used inside a TypedDict
```
Comment (JukkaL) [inline test-data/unit/check-typeddict.test:4065]: Similar to above, should we test the functional form as well?

**Review by JukkaL — COMMENTED** [review]
Thanks for the PR! Left questions about functional forms.

**Comment by sobolevn** [comment]
Thanks, addressed! 👍

**Pull request merged**